### ISSUE TITLE
sfdisk: bug in create gpt new partitions from text file on smaller drive.

### ISSUE THREAD

**Issue by ivanmara555** [util-linux/util-linux#3186]
Let's assume we have 1T drive with 100G partition.

Dump with:
`sfdisk --dump /dev/vda > win_0.gpt.part.backup.txt`
will create text file:
```
label: gpt
label-id: 5F4CF436-83E0-47B3-93E4-122F36E0A131
device: /dev/vda
unit: sectors
first-lba: 34
last-lba: 2147483614
sector-size: 512

/dev/vda1 : start=        2048, size=   209715200, type=EBD0A0A2-B9E5-4433-87C0-68B6B72699C7, uuid=D49C17A7-F6C6-4FCB-9A6E-0571416539BF, name="Microsoft basic data"

```
Now i want to create partitions from created file on smaller 500G drive:
```
sfdisk /dev/vda < win_0.gpt.part.backup.txt
# or
sfdisk --force /dev/vda < win_0.gpt.part.backup.txt
```

The commands ends with an error:
```
>>> Last LBA specified by script is out of range.
Last LBA specified by script is out of range.
Failed to apply script headers, disk label not created: Numerical result out of range
```

But if i manually delete  string `last-lba: 2147483614` from backup text file the commands works!

The bug is that sfdisk does not create gpt partitions from text file on smaller drive even if it is specified `--force` flag but although it may.

p.s.: For creating partitions on smaller drive i use: `sgdisk --load-backup=win_0.gpt.part.backup.bin /dev/vda` but prefer sfdisk.

**Comment by karelzak** [comment]
Good idea, I think we can ignore `last-lba` on --force as this setting is unimportant.

**Comment by ivanmara555** [comment]
> Good idea, I think we can ignore `last-lba` on --force as this setting is unimportant.

As far as I understand from the current documentation (man sfdisk say: last-lba: Specify the last usable sector for GPT partitions. This header is ignored if the script and device sector size differ.) sfdisk must work w/o `--force` option. For example in the example above 100Gb partition does not depend on the disk size (disk much greater) and can be create anyway, therefore we can ignore `last-lba` header and the `--force` option is not needed in any case.

**Comment by karelzak** [comment]
I considered ignoring the `last-lba` header, but I'll keep it for now. The header allows for defining values within the on-disk GPT header; if anyone does not like it, he does not use it (or use --force). The header is used in the dump to allow the creation of a copy of the GPT on-disk label.

Closing, fixed by 4877d7f61a72e46aba3fc52f347d1cd74124dba5.
